ml4gw/waveforms/phenom_p.py

```diff
@@ -0,0 +1,731 @@
+            + (575 * chip2 * m2_4) / (1536.0 * mtot4 * eta)
+            + (39695 * eta) / 86016.0
+            + (1615 * dm2 * eta) / (28672.0 * m2_2)
+            - (265 * dm * eta) / (14336.0 * m2)
+            + (955 * eta2) / 576.0
+            + (15 * dm3 * eta2) / (1024.0 * m2_3)
+            + (35 * dm2 * eta2) / (256.0 * m2_2)
+            + (2725 * dm * eta2) / (3072.0 * m2)
+            - (15 * dm * m2 * PI * chil) / (16.0 * mtot2 * eta)
+            - (35 * m2_2 * PI * chil) / (16.0 * mtot2 * eta)
+            + (15 * chip2 * dm * m2_7 * chil2) / (128.0 * mtot8 * eta4)
+            + (35 * chip2 * m2_8 * chil2) / (128.0 * mtot8 * eta4)
+            + (375 * dm2 * m2_2 * chil2) / (256.0 * mtot4 * eta)
+            + (1815 * dm * m2_3 * chil2) / (256.0 * mtot4 * eta)
+            + (1645 * m2_4 * chil2) / (192.0 * mtot4 * eta)
+        )
+
+        angcoeffs["epsiloncoeff1"] = -0.18229166666666666 - (5 * dm) / (
+            64.0 * m2
+        )
+        angcoeffs["epsiloncoeff2"] = (-15 * dm * m2 * chil) / (
+            128.0 * mtot2 * eta
+        ) - (35 * m2_2 * chil) / (128.0 * mtot2 * eta)
+        angcoeffs["epsiloncoeff3"] = (
+            -1.7952473958333333
+            - (4555 * dm) / (7168.0 * m2)
+            - (515 * eta) / 384.0
+            - (15 * dm2 * eta) / (256.0 * m2_2)
+            - (175 * dm * eta) / (256.0 * m2)
+        )
+        angcoeffs["epsiloncoeff4"] = (
+            -(35 * PI) / 48.0
+            - (5 * dm * PI) / (16.0 * m2)
+            + (5 * dm2 * chil) / (16.0 * mtot2)
+            + (5 * dm * m2 * chil) / (3.0 * mtot2)
+            + (2545 * m2_2 * chil) / (1152.0 * mtot2)
+            + (2035 * dm * m2 * chil) / (21504.0 * mtot2 * eta)
+            + (2995 * m2_2 * chil) / (9216.0 * mtot2 * eta)
+        )
+        angcoeffs["epsiloncoeff5"] = (
+            4.318908476114694
+            + (27895885 * dm) / (2.1676032e7 * m2)
+            + (39695 * eta) / 86016.0
+            + (1615 * dm2 * eta) / (28672.0 * m2_2)
+            - (265 * dm * eta) / (14336.0 * m2)
+            + (955 * eta2) / 576.0
+            + (15 * dm3 * eta2) / (1024.0 * m2_3)
+            + (35 * dm2 * eta2) / (256.0 * m2_2)
+            + (2725 * dm * eta2) / (3072.0 * m2)
+            - (15 * dm * m2 * PI * chil) / (16.0 * mtot2 * eta)
+            - (35 * m2_2 * PI * chil) / (16.0 * mtot2 * eta)
+            + (375 * dm2 * m2_2 * chil2) / (256.0 * mtot4 * eta)
+            + (1815 * dm * m2_3 * chil2) / (256.0 * mtot4 * eta)
+            + (1645 * m2_4 * chil2) / (192.0 * mtot4 * eta)
+        )
+        return angcoeffs
+
+    def FinalSpin_inplane(
+        self,
+        m1: TensorType,
+        m2: TensorType,
+        chi1_l: TensorType,
+        chi2_l: TensorType,
+        chip: TensorType,
+    ) -> TensorType:
+        M = m1 + m2
+        eta = m1 * m2 / (M * M)
+        eta2 = eta * eta
+        q_factor = torch.where(m1 >= m2, m1 / M, m2 / M)
```

Comment:
Can we avoid the `.where` if we switch to using chirp_mass and mass_ratio?

Reply:
this is my bad, the [ripple version](https://github.com/tedwards2412/ripple/blob/c451e87cd3130286fb2bd32c4148507ea05a3dff/src/ripplegw/waveforms/IMRPhenomPv2_utils.py#L354), which this code is based on assumes a convention of m1 being greater than m2. I copied this function (FinalSpin_inplane) from the LAL source code, which actually flips the variables if m1<m2. during my checks of the waveform, I found it didnt give correct results for m2>m1 due to other m1>m2 assumptions from the ripple code. I've made it such that the code doesn't decide values based on m1 and m2 now.